### PR TITLE
[Hardware][AMD][CI][Bugfix] Fix AMD Quantization test group

### PR DESCRIPTION
**Pull request by mawong-amd**
## Purpose
This PR fixes all remaining failures in the AMD Quantization Tests group. Fixes https://github.com/orgs/vllm-project/projects/39/views/1?pane=issue&itemId=141650694&issue=vllm-project%7Cvllm%7C29525.

## Test Plan
CI will run the following test command on AMD hardware
`VLLM_TEST_FORCE_LOAD_FORMAT=auto pytest -v -s quantization/ --ignore quantization/test_blackwell_moe.py`.

## Test Result
All tests should pass.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit cbbbcaf88499865607b199c3513a0d26923be392. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Improves AMD quantization reliability across CI and runtime by aligning tests with ROCm capabilities and tightening platform checks.
> 
> - CI: `Quantization Test` now mirrors on `amdquantization` hardware in `.buildkite/test-amd.yaml`
> - Tests: add ROCm/CUDA-aware skips/params and expectations
>   - Use `current_platform.fp8_dtype()` and accept HIP paths; gate Marlin use to CUDA; adjust model IDs and CPU offload cases
>   - Simplify/PTPC FP8 test to ROCm bf16/FP8 paths; platform-aware expected quant types in `test_configs`
>   - `is_quant_method_supported` now verifies via `current_platform.verify_quantization`
> - Kernels/quantization runtime fixes
>   - Triton scaled-mm: convert per-tensor scales to per-channel for fused modules
>   - PTPC FP8: clearer dtype assertions; support pre-FP8-serialized weights or on-load quantization; keep `weight_scale`
>   - FP8 utils: normalize e4m3fn→e4m3fnuz unconditionally; dispatch respects ROCm FP8 dtype
>   - Quantization registry marks custom methods as supported on the current platform; ROCm declares `awq_marlin`/`gptq_marlin` aliases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit cbbbcaf88499865607b199c3513a0d26923be392. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit 9b77a63de6ebf1e1d9fad28001aeefdcb8c11743. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Improves AMD/ROCm quantization reliability across CI and runtime by aligning tests with ROCm capabilities and tightening platform checks.
> 
> - CI: `Quantization Test` mirrors on `amdquantization` hardware in `.buildkite/test-amd.yaml`
> - Tests: platform-aware skips/params and expectations
>   - Gate Marlin-only paths to CUDA; use `current_platform.fp8_dtype()` and accept HIP paths; adjust model IDs and CPU offload cases
>   - Simplify `ptpc_fp8` to ROCm bf16/FP8; platform-conditional expected types in `test_configs`; validate support via `current_platform.verify_quantization`
> - Kernels/runtime: small fixes for correctness/compat
>   - `TritonScaledMMLinearKernel`: convert per-tensor scales to per-channel for fused modules
>   - `PTPCFp8LinearMethod`: clearer dtype checks; support pre-FP8-serialized weights or on-load quantization; preserve `weight_scale`
>   - FP8 utils: early-return normalize e4m3fn→e4m3fnuz and respect ROCm FP8 dtype; allow HIP forward in `QuantFP8`
>   - Quant registry marks custom methods as supported; ROCm declares `awq_marlin`/`gptq_marlin` aliases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 9b77a63de6ebf1e1d9fad28001aeefdcb8c11743. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> <sup>[Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) is generating a summary for commit bba5d2e5dac54251008ddf925af5d6f6dbd04176. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Improves AMD/ROCm quantization stability across CI and runtime.
> 
> - CI: `Quantization Test` now mirrors on `amdquantization` hardware in `.buildkite/test-amd.yaml`
> - Tests: add ROCm/CUDA-aware skips/params and expectations; gate Marlin-only paths to CUDA; use `current_platform.fp8_dtype()`; accept HIP forward in FP8 ops; update model IDs and CPU offload cases; simplify `ptpc_fp8` to ROCm BF16/FP8; platform-conditional expected types in `test_configs`; support checks via `current_platform.verify_quantization`
> - Runtime/kernel fixes: Triton scaled-mm converts per-tensor scales to per-channel for fused modules; `PTPCFp8LinearMethod` refines dtype assertions and supports pre-FP8-serialized weights; quantization registry marks custom methods as supported; ROCm declares `awq_marlin`/`gptq_marlin` aliases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit bba5d2e5dac54251008ddf925af5d6f6dbd04176. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

---

> [!NOTE]
> Improves AMD/ROCm quantization reliability across CI and runtime.
> 
> - CI: `Quantization Test` now mirrors on `amdproduction` in `.buildkite/test-amd.yaml`
> - Tests: platform-aware skips/params and expectations
>   - Gate Marlin-only options to CUDA; use `current_platform.fp8_dtype()`; accept HIP forward in FP8 ops; update CPU offload model
>   - `ptpc_fp8` test simplified to ROCm BF16/FP8 and marked deprecated path via `allow_deprecated_quantization`
>   - `test_configs` expects `awq_marlin/gptq_marlin` only on CUDA; validate support with `current_platform.verify_quantization`
> - Runtime/kernel fixes
>   - Triton scaled-mm: convert per-tensor scales to per-channel for fused modules
>   - PTPC FP8: refined dtype assertions; support pre-FP8-serialized weights or on-load quantization; retain `weight_scale`
>   - Quantization registry: custom methods marked supported; ROCm declares `awq_marlin`/`gptq_marlin` aliases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0cdf86c8dd2a48300b7ac47ae41e51e760b6c0d4. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/utils/w8a8_utils.py [hunk 1]
```diff
@@ -500,7 +500,11 @@ def normalize_e4m3fn_to_e4m3fnuz(
     weight_scale: torch.Tensor,
     input_scale: torch.Tensor | None = None,
 ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor | None]:
-    assert weight.dtype == torch.float8_e4m3fn
```
Comment (tjtanaa) [inline vllm/model_executor/layers/quantization/utils/w8a8_utils.py:503]: This function is not idempotent, 

If apply it on matrix A twice, A (fn) -> B (fnuz) -> C (fnuz), The scale of C is 2x of B. So this assertion is important to ensure that we don't apply it twice. We should try to fix the inconsistency of `It may be the case that the incoming weight is already of dtype torch.float8_e4m3fnuz, but actually contains data in torch.float8_e4m3fn` before calling this function.

File: vllm/model_executor/layers/quantization/ptpc_fp8.py [hunk 2]
```diff
@@ -105,19 +105,25 @@ def __init__(self, quant_config: PTPCFp8Config):
...
+            )
+
+            # Update the layer with the new values.
+            layer.weight = Parameter(
+                qweight.t(), requires_grad=False
+            )  # Pretranspose the weight
+            layer.weight_scale = Parameter(weight_scale, requires_grad=False)
+        else:
+            assert layer.weight.data.dtype == current_platform.fp8_dtype()
+            assert getattr(layer, "weight_scale", None) is not None
```
[the remark this answers is not in the document]
Comment (mawong-amd) [reply vllm/model_executor/layers/quantization/ptpc_fp8.py:124]: Not necessary. The necessary transposes for the `else` branch have already been completed in that case.

File: vllm/model_executor/layers/quantization/utils/w8a8_utils.py [hunk 3]
```diff
@@ -500,7 +500,11 @@ def normalize_e4m3fn_to_e4m3fnuz(
...
+    # If weight is already in torch.float8_e4m3fnuz format,
+    # do nothing
+    if weight.dtype == torch.float8_e4m3fnuz:
+        return weight, weight_scale, input_scale
```
[the remark this answers is not in the document]
Comment (mawong-amd) [reply vllm/model_executor/layers/quantization/utils/w8a8_utils.py]: Portion reverted.

File: vllm/model_executor/layers/quantization/utils/w8a8_utils.py (hunk 1 above)
Comment (mawong-amd) [reply vllm/model_executor/layers/quantization/utils/w8a8_utils.py:503]: These changes have been reverted, because after some further thought, I don't believe they achieve the intended effect.

The inconsistency arises because of weight reloading as implemented [here](https://github.com/vllm-project/vllm/pull/28480). In particular, we might load e4m3fn weights, process them to e4m3fnuz, and then try to reload e4m3fn weights. The issue then is that we would need to 1. copy e4m3fn data to an e4m3fnuz tensor and 2. ensure this data is normalized as usual.

After some further thought, I don't believe that there is a nice way to do this outside of patching the weight loader for e4m3fnuz weights to do the normalization on the fly. Such an approach would also require detecting whether we've performed the necessary *2 for the associated scales afterwards in `process_weights_after_loading`. I think the changes  to enable FP8 e4m3fn weight reloading on e4m3fnuz architectures go beyond the scope of this PR, so I've removed them.

File: vllm/platforms/rocm.py [hunk 4]
```diff
@@ -170,7 +170,9 @@ class RocmPlatform(Platform):
 
     supported_quantization: list[str] = [
         "awq",
+        "awq_marlin",  # will be overwritten with awq
         "gptq",
+        "gptq_marlin",  # will be overwritten with gptq
```
[the remark this answers is not in the document]
Comment (mawong-amd) [reply vllm/platforms/rocm.py:175]: These marlin quantizations are overwritten with their non-marlin equivalents on ROCm, which run as expected. If they were not added here, then this overwriting would not even take place (because these would be deemed unsupported earlier).

[the remark this answers is not in the document]
Comment (mawong-amd) [reply vllm/platforms/rocm.py:175]: These marlin quantizations are overwritten with their non-marlin equivalents on ROCm, which run as expected. If they were not added here, then this overwriting would not even take place (because these would be deemed unsupported earlier).

Resolving as duplicate.

File: tests/quantization/test_cpu_offload.py [hunk 5]
```diff
@@ -66,7 +66,7 @@ def test_cpu_offload_compressed_tensors(monkeypatch):
     monkeypatch.setenv("VLLM_TEST_FORCE_LOAD_FORMAT", "auto")
     # Test wNa16
     compare_two_settings(
-        "nm-testing/tinyllama-oneshot-w4a16-channel-v2",
+        "nm-testing/Qwen1.5-MoE-A2.7B-Chat-quantized.w4a16",
```
Comment (tjtanaa) [inline tests/quantization/test_cpu_offload.py:69]: @mawong-amd Any model should be fine. But may I know why change the model?

Comment (mawong-amd) [reply tests/quantization/test_cpu_offload.py:69]: I'm running into a segfault in [this Conch Triton kernel](https://github.com/stackav-oss/conch/blob/e2c5ecfe341574114fef0218dedb1dd0da1e71f2/conch/kernels/quantization/gemm.py#L220) while on the original model.
The segfault might be worth looking into separately, but for the purposes of this PR, it seemed to me that the purposes of this test are still satisfied by using the replacement model.

**Review by tjtanaa — APPROVED** [review]
LGTM
